sphinx_immaterial/cpp_domain_fixes.py

```diff
@@ -708,12 +725,12 @@ def add_replacement(
 
 
 def _add_parameter_documentation_ids(
-    env,
+    env: sphinx.environment.BuildEnvironment,
     obj_content: sphinx.addnodes.desc_content,
-    sig_param_nodes_for_signature: List[Dict[str, docutils.nodes.Element]],
-    symbols,
-    domain_module,
-    starting_id_version,
+    sig_param_nodes_for_signature: List[Dict[str, sphinx.addnodes.desc_parameter]],
```

Comment:
These aren't all `desc_parameter` nodes because this is also used with template parameters that don't use that node.

Reply:
I actually tracked this list down manually 
https://github.com/jbms/sphinx-immaterial/blob/4460d99c71b95a7ea0bbf7692d027f9e883dcde3/sphinx_immaterial/cpp_domain_fixes.py#L690-L696

I don't see how `add_replacement()` changes the `desc_parameter` to anything else.

Reply:
For that call the type is `desc_parameter` but there is a second call to `add_replacement` below for the template parameter case:
https://github.com/jbms/sphinx-immaterial/blob/4460d99c71b95a7ea0bbf7692d027f9e883dcde3/sphinx_immaterial/cpp_domain_fixes.py#L706